shell/platform/darwin/macos/framework/Source/FlutterViewController.mm

```diff
@@ -243,6 +249,12 @@ - (void)viewWillAppear {
   }
 }
 
+- (void)viewWillDisappear {
+  // Per Apple's documentation, it is discouraged to call removeMonitor: in dealloc, and it's
+  // recommended to be called earlier in the lifecycle.
+  [NSEvent removeMonitor:_keyUpMonitor];
```

Comment:
nil the monitor ivar here to avoid any possible confusion where it's present but not listening.

Reply:
Done